src/Hazelcast.Net.Linq.Async/HMapExtension.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+// http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Linq;
+using Hazelcast.DistributedObjects;
+using Hazelcast.DistributedObjects.Impl;
+
+namespace Hazelcast.Linq
+{
+    public static class HMapExtension
+    {
+        /// <summary>
+        /// Extension of <see cref="IHMap"/> for LINQ functionalities. It provides IAsyncQueryable part of the <see cref="IHMap"/>.  
+        /// </summary>
+        /// <param name="hMap"><see cref="IHMap"/> to be queried.</param>
+        /// <typeparam name="TKey">Type of key of the <see cref="IHMap"/></typeparam>
+        /// <typeparam name="TValue">Type of value of the <see cref="IHMap"/></typeparam>
+        /// <returns></returns>
+        public static IAsyncQueryable<HKeyValuePair<TKey, TValue>> AsAsyncQueryable<TKey, TValue>(this IHMap<TKey, TValue> hMap)
```

Comment:
So we've probably discussed this but I forgot - why cannot `IHMap` directly implement the interface and why do ppl have to do the `AsAsyncQueryable` thing? Can we document the reason?

Reply:
Extension method provides Linq abilities from an another package. Otherwise, ÌHMap` should implement the query provider interfaces, so we cannot separate Linq into a different package.

Reply:
totally make sense. maybe at the top of the class, so we remember:
```
// async LINQ is provided as an extension package - so IHMap cannot directly expose async LINQ - so we have to
// go with an extension method to map it to something async-queryable
```

Reply:
Fixed at https://github.com/hazelcast/hazelcast-csharp-client/pull/752/commits/4506e1ba5fd0bae9f9c763cf789a39e0f652006f